pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/IndexLoadingConfig.java

```diff
@@ -304,6 +306,13 @@ public void setReadMode(ReadMode readMode) {
   public Map<String, Map<String, String>> getColumnProperties() {
     return _columnProperties;
   }
+  public int getRangeIndexVersion() {
```

Comment:
@Jackie-Jiang it's not 100% clear to me if this is the right place to be configuring this, please let me know your thoughts.

Reply:
`IndexLoadingConfig` is not exposed to the user, but initialized from the table config. We can put this index version into the `IndexingConfig` which is a sub-config under the `TableConfig` so that user can specify the range index version they want to generate.

Reply:
I originally put this in `IndexingConfig` but unfortunately the table config was null in some unit tests, which looks like an error. I'll move it back.